### PR TITLE
`BoxControl`: stop using `UnitControl`'s deprecated `unit` prop

### PR DESCRIPTION
**Pull request by ciampo**
## What?

Related to #39503 , this PR refactors the `BoxControl` component to avoid using the deprecated `unit` prop from the `UnitControl` component.

## Why?

The `unit` prop is marked as deprecated, the component's docs recommend passing the unit directly through the `value` prop

## How?

Refactor the code to pass the unit directly through the `value` prop

## Testing Instructions

- Tests pass
- Component behaves as expected in Storybook and in the editor.

### CODE REVIEW

**Review by andrewserong — APPROVED** [review]
This is testing nicely for me in the editor (e.g. padding controls on the group block, and for axial controls, padding controls on the button block) and in Storybook. It appears to behave exactly the same as on `trunk` as far as I can tell! 👍

The concatenated unit value appears to be present on trunk for `0` based values (e.g. `0px`) so that doesn't appear to be different here, either.

LGTM! Since there's a fair bit going on in this component, it might be worth getting a second pair of eyes before landing the PR, just as a confidence check 🙂

**Comment by ciampo** [comment]
> Since there's a fair bit going on in this component, it might be worth getting a second pair of eyes before landing the PR, just as a confidence check 🙂

Couldn't agree more. I actually spent more than I wanted on this refactor exactly because I didn't know the component very well and was afraid on introducing a regression.

Let's wait for a second pair of eyes to review this

File: packages/components/src/box-control/input-controls.js
Comment (ciampo) [inline packages/components/src/box-control/input-controls.js:101]: For ease of review: the `unit` prop here was removed, and instead the parsing and potentially overriding of the unit is applied to the `value` prop

File: packages/components/src/box-control/axial-input-controls.js
Comment (ciampo) [inline packages/components/src/box-control/axial-input-controls.js:123]: For ease of review: the `unit` prop here was removed, and instead the parsing and potentially overriding of the unit is applied to the `value` prop

File: packages/components/src/box-control/all-input-control.js
Comment (ciampo) [inline packages/components/src/box-control/all-input-control.js:36]: For ease of review: the `unit` prop below was removed. As a consequence, the `allUnitFallback` variable was also removed. The logic used for computing `allUnitFallback` has been moved and "embedded" in the `getAllValue` utility function

File: packages/components/src/box-control/utils.js
Comment (mirka) [inline packages/components/src/box-control/utils.js:81]: Verbosity FTW

**Review by mirka — APPROVED** [review]
Changes make sense to me, and the inline comments were helpful. Thanks!

**Pull request merged**
